### PR TITLE
Gitkraken cli

### PR DESCRIPTION
**Pull request by alexs-gitK**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

**Comment by krehel** [comment]
Can you please explain this PR, as the `gitkraken-cli` appears to be a single binary without shell completions.

**Comment by alexs-gitK** [comment]
> Can you please explain this PR, as the `gitkraken-cli` appears to be a single binary without shell completions
@krehel 
The internal PR is still in review, once that's complemented then the completion should be in the zip.  But are you concerned with what’s inside our zip?

**Comment by krehel** [comment]
Concern is not the intent, but this should be opened when the files do exist in the release.

There is a `binary` stanza directive that should likely be used for the shell completions, which will cause failure if the source files don't exist.

An example of the implementation can be seen in https://github.com/Homebrew/homebrew-cask/blob/f9b57971598ffc356b8ce6d25538df2818273cb8/Casks/a/alacritty.rb.

The change to add completions should probably be a few lines.

**Comment by bevanjkay** [comment]
> Your example is not applicable as that example is meant for a traditional MacOS application that has its resources embedded within.

The `binary` stanza can, and should still be used in the same was as the example above, and is preferred. 


Regarding your other points, you are welcome to submit a PR to suggest any improvements you can see, but please adjust your tone in your communications.

File: Casks/g/gitkraken-cli.rb [hunk 1]
```diff
@@ -12,5 +12,43 @@
...
+
+  uninstall_postflight do
+    FileUtils.rm_f "#{HOMEBREW_PREFIX}/etc/bash_completion.d/gk"
+    FileUtils.rm_f "#{HOMEBREW_PREFIX}/share/fish/vendor_completions.d/gk.fish"
+    FileUtils.rm_f "#{HOMEBREW_PREFIX}/share/zsh/site-functions/_gk"
+  end
+
   zap trash: "~/.gitkraken"
+
+  caveats <<~EOS
```
Comment (bevanjkay) [inline Casks/g/gitkraken-cli.rb:32]: Is the information in the caveat specific to Homebrew? If these instructions would usually be required (if installing in another way) then we shouldn't be adding a caveat here.

**Comment by alexs-gitK** [comment]
> > Your example is not applicable as that example is meant for a traditional MacOS application that has its resources embedded within.
> 
> The `binary` stanza can, and should still be used in the same was as the example above, and is preferred.
> 
> Regarding your other points, you are welcome to submit a PR to suggest any improvements you can see, but please adjust your tone in your communications.

@bevanjkay, 

Regarding your comment about my tone, I did not use any tone in my communication. I was merely expressing that the Homebrew-cask ecosystem needs to be clearer about the code expectations upfront about do's and don'ts. Moreover, I've noticed that many examples focus on traditional macOS applications that package resources within a .app bundle and that the binary stanza was primarily intended for use with .app bundles. Our command-line interface (CLI) application tool does not fit this mold, which has caused some confusion regarding the correct approach for including non-bundled binaries and their associated completion scripts.

Given this context, I've adjusted our cask submission as follows to better comply with Homebrew Cask standards for CLI tools:
```ruby
binary "gk" binary "gk.bash", 
target: "#{HOMEBREW_PREFIX}/etc/bash_completion.d/gk"
binary "_gk", 
target: "#{HOMEBREW_PREFIX}/share/zsh/site-functions/_gk" 
binary "gk.fish", 
target: "#{HOMEBREW_PREFIX}/share/fish/vendor_completions.d/gk.fish"
```
Could you please confirm if this adjustment aligns with Homebrew Cask's expectations for CLI applications distributed without a .app bundle?

Switching to the binary stanzas requires the specified files to be present in the ZIP archive. Our internal PR, which includes these completion scripts in the archive, is still pending. Given this, would it be preferable to close this PR temporarily and reopen it once our internal updates are live, or is there a protocol for placing the PR on hold in the interim?

**Comment by p-linnane** [comment]
The `caveats` section here is not appropriate for Homebrew, since understanding how to use shell completions is not a Homebrew-specific problem.

Please be respectful of maintainers. Open PR's with proposed fixes instead of complaining about "lousy code that is buggy".

**Pull request closed without merging**